### ISSUE TITLE
setScaleType has no effect

### ISSUE THREAD

**Issue by artworkad** [brianwernick/ExoMedia#465]
- ✅ I have verified there are no duplicate active or recent bugs, questions, or requests

###### Include the following:
 - ExoMedia version: `4.0.2`
 - Device OS version: `5.1`
 - Devide Manufacturer: `LG`
 - Device Name: `NEXUS 4`
 
###### Reproduction Steps
Created fresh fragment and put following code in onStart:
```
video_view.setVideoURI(Uri.parse("/android_asset/onboarding.mp4"))
video_view.setMeasureBasedOnAspectRatioEnabled(false)
video_view.setScaleType(ScaleType.CENTER_CROP)
video_view.setOnPreparedListener {
     video_view.start()
 }
video_view.setOnCompletionListener {
    video_view.restart()
}
```

Layout:

```
<?xml version="1.0" encoding="utf-8"?>
<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:background="@color/white">

    <com.devbrackets.android.exomedia.ui.widget.VideoView
        android:id="@+id/video_view"
        android:layout_width="match_parent"
        android:layout_height="match_parent" />

</FrameLayout>
```

###### Expected Result

The video should fill the entire screen and be cropped. The aspect ration should not change.

###### Actual Result

The video fills the screen, however there is a black padding at the bottom, which is not coming from the layout. I added a white background to my layout to check this. So the black space must come from within the video view. Besides that the aspect ration of the video changed.

###### Description

So I was looking for a way to have the video just scale and crop to fill the entire screen. But setting the scale type has not the desired effect. The video has full height but not full width (white space on left and right). Then I stumbled upon this issue https://github.com/brianwernick/ExoMedia/issues/164 and set

`video_view.setMeasureBasedOnAspectRatioEnabled(false)`

which even created another problem (change of aspect ratio)

**Comment by brianwernick** [comment]
This sounds like an issue with the `SurfaceView` implementation (default), if this is something you need now then the `TextureView` backing should work correctly (add `app:useTextureViewBacking="true"`)

**Comment by ArtworkAD** [comment]
@brianwernick adding app:useTextureViewBacking fixed the issue

**Comment by caleroalbelo90** [comment]
Did you already have a solution about setScaleType has no effect...i'm getting the same problem..

**Comment by dtvcfinc** [comment]
@brianwernick thanks, your solution works, but is there any plan to fix the surfaceview implementation?  From [google documentation:](https://source.android.com/devices/graphics/arch-tv#surface_or_texture) For a full-screen video player, or any other application that is effectively just UI elements layered on top of video, SurfaceView offers much better performance.

**Comment by brianwernick** [comment]
Yeah, the performance improvements in the SurfaceView vs the TextureView is why it's used as the default backing. I definitely do plan on fixing this, I just haven't had much time to work on it yet; once I get the 2.0 release out for PlaylistCore I should be able to focus on bug fixes/improvements in ExoMedia

**Comment by Kisty** [comment]
I've got a problem where on a Sony Z3 compact tablet, without `app:useTextureViewBacking="true"`, the video is stretched and the right part is cut off. Tried using `app:measureBasedOnAspectRatio="true"` noting the change in https://github.com/brianwernick/ExoMedia/commit/81b2fe8b742eefe3cd940ba84e9f82d9daf5f2d2. I'll see what version it broke in. Using ExoMedia 4.0.3.

![screenshot_20170904-124109](https://user-images.githubusercontent.com/3243752/30025095-d7e6a1f2-916e-11e7-95bd-c10eea3ce929.png)
with  `app:useTextureViewBacking="true"`

![screenshot_20170904-124149](https://user-images.githubusercontent.com/3243752/30025056-acc7219a-916e-11e7-877c-e96fa3444ae6.png)
without `app:useTextureViewBacking="true"`
